rfcs/sexp.md

```diff
@@ -0,0 +1,104 @@
+# Add a `Sexp` module to the standard library
+
+I propose to add a `Sexp` module to the standard library.
+The goal is twofold:
+
+1. Provide a common type definition that can be shared
+   among libraries that deal with s-expressions.
+
+2. Promote the use of s-expressions as a lightweight
+   human-readable serialization format.
+
+The module would include at least the following definitions:
+
+```
+type t =
+  | Atom of string
+  | List of t list
+
+val equal: t -> t -> bool
+
+val compare: t -> t -> int
+
+val pp: Format.formatter -> t -> unit
+
+val to_string: t -> string
+
+val of_string: string -> t option
+```
+
+**Question:** The above functions will pretty-print their arguments.  Should
+"compact" variants (minimum amount of whitespace) also be included?
+
+## Support in other modules of the standard library
+
+In addition, functions
+
+- `to_sexp : t -> Sexp.t`
+- `of_sexp_opt: Sexp.t -> t option`
+- `of_sexp: Sexp.t -> t`
+
+would be added to all existing modules of the stdlib for which it makes sense:
```

Comment:
I find this to be a bit ad-hoc and no so forward looking. There are quite a few other formats you may want to codec with and this is not going to work for these. Not to mention providing `Formatter -> t -> unit` functions for all of these types.

I think it would be more useful to research and provide a generic type representation, add it to the stdlib and have in the `Sexp` module, say  `Sexp.to_type : 'a Typ.t -> string  -> ('a, error) result` and `Sexp.of_type : 'a Typ.t -> 'a -> Sexp.t` functions.